internal/state/validation.go

```diff
@@ -119,6 +119,15 @@ func validateBlock(state State, block *types.Block) error {
 				state.LastBlockTime,
 			)
 		}
+		if !state.ConsensusParams.PBTS.PBTSEnabled(block.Height) && !isBlockSync {
```

Comment:
I see. I think I remember. Was the reason for skipping this in blocksync that, since the blocks are light-client-validated, we assume the data is correct? (as we make the same assumption with the `app_hash`?

If that's the case, @cason, I think the answer to your question is that, by skipping the unneeded check, we are being faster.

Reply:
Exactly. The whole test had been removed as not needed when using PBTS, but since we need to support BFT time, it was brought back.
However during the discussion we concluded that even with BFT Time the test is not needed during block sync, since the timestamp has been verified when it was decided and we check the block hash (which includes the time) that this is the one decided.

Reply:
OK, we changed our mind yesterday and decided to go for _not_ modifying the checks we do when in BFT Time w.r.t. the checks performed in `v0.38.x`